src/daft-dsl/src/expr/mod.rs

```diff
@@ -1589,14 +1589,16 @@ pub fn estimated_selectivity(expr: &Expr, schema: &Schema) -> f64 {
             let right_selectivity = estimated_selectivity(right, schema);
             match op {
                 // Fixed selectivity for all common comparisons
-                Operator::Eq => 0.05,
-                Operator::EqNullSafe => 0.05,
-                Operator::NotEq => 0.95,
-                Operator::Lt | Operator::LtEq | Operator::Gt | Operator::GtEq => 0.5,
+                Operator::Eq => 0.2,
+                Operator::EqNullSafe => 0.2,
+                Operator::NotEq => 0.8,
+                Operator::Lt | Operator::LtEq | Operator::Gt | Operator::GtEq => 0.3,
 
                 // Logical operators with fixed estimates
-                // P(A and B) = P(A) * P(B)
-                Operator::And => left_selectivity * right_selectivity,
+                // Use the minimum selectivity of the two operands for AND
+                // This is a more conservative estimate than the product of the two selectivities,
+                // because we cannot assume independence between the two operands.
+                Operator::And => left_selectivity.min(right_selectivity),
```

Comment:
Another option here is to apply a penalty to the multiplication so we can choose some new selectivity from `[left * right, left.min(right)]` if we wanted more tuning.